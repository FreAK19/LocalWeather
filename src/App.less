@bg-color: rgba(38, 50, 56, 1);
@bg-color-viewport: #d1d1d1;
/*! normalize.css v3.0.2 | MIT License | git.io/normalize */

/**
 * 1. Set default font family to sans-serif.
 * 2. Prevent iOS text size adjust after orientation change, without disabling
 *    user zoom.
 */

html {
  font-family: sans-serif; /* 1 */
  -ms-text-size-adjust: 100%; /* 2 */
  -webkit-text-size-adjust: 100%; /* 2 */
}

/**
 * Remove default margin.
 */

body {
  margin: 0;
}

/* HTML5 display definitions
   ========================================================================== */

/**
 * Correct `block` display not defined for any HTML5 element in IE 8/9.
 * Correct `block` display not defined for `details` or `summary` in IE 10/11
 * and Firefox.
 * Correct `block` display not defined for `main` in IE 11.
 */

article,
aside,
details,
figcaption,
figure,
footer,
header,
hgroup,
main,
menu,
nav,
section,
summary {
  display: block;
}

/* Links
   ========================================================================== */

/**
 * Remove the gray background color from active links in IE 10.
 */

a {
  background-color: transparent;
}

a:active,
a:hover {
  outline: 0;
}

b,
strong {
  font-weight: bold;
}

/**
 * Address variable `h1` font-size and margin within `section` and `article`
 * contexts in Firefox 4+, Safari, and Chrome.
 */

h1 {
  font-size: 2em;
  margin: 0.67em 0;
}
/* Tables
   ========================================================================== */

/**
 * Remove most spacing between table cells.
 */

table {
  border-collapse: collapse;
  border-spacing: 0;
}

td,
th {
  padding: 0;
}

@import "asset/css/pe-icon-set-weather.css";

/* Base */

body {
  font-family: 'Roboto', Tahoma sans-serif;
  font-weight: 400;
}

a {
  text-decoration: none;
}

/* Box model */

*,
*::after,
*::before {
  -webkit-box-sizing: border-box;
  -moz-box-sizing: border-box;
  box-sizing: border-box;
}

#app {
  width: 100vw;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  background: @bg-color;

  @media screen and (max-width: 640px) {
    & {
      background: @bg-color-viewport;
    }
  }
}

.main {
  width: 100vw;
  height: 100vh;
  max-width: 768px;
  display: flex;
  align-items: center;
}