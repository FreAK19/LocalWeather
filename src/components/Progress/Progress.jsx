import React from 'react';
import CircularProgress from 'material-ui/CircularProgress/CircularProgress';
import PropTypes from 'prop-types';
import './Progress.less';

const colorSpinner = '#485';

const Progress = ({ show }) =>
  show ? (
    <div className="spinner">
      <div className="spinner__box">
        <CircularProgress size={60} color={colorSpinner} />
      </div>
    </div>
  ) : null;

Progress.propTypes = { show: PropTypes.bool.isRequired };

export default Progress;
