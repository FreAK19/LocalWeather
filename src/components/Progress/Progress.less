@bg-color: rgba(0, 188, 212, 0.48);

.spinner {
  position: absolute;
  top: 0;
  right: 0;
  left: 0;
  bottom: 0;
  background: @bg-color;
  z-index: 999;

  &__box {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}