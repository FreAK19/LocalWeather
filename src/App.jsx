import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import './App.less';
import Progress from './components/Progress/Progress';
import WeatherContainer from './components/WeatherContainer/WeatherContainer';

export default class App extends Component {
  state = {
    error: false,
    data: {
      main: {
        temp: 0,
        pressure: 0,
        humidity: 0,
        temp_min: 0,
        temp_max: 0
      },
      name: '',
      wind: {
        speed: 0
      },
      weather: [
        {
          id: 800,
          main: 'Clear',
          description: 'Weather Status'
        }
      ]
    },
    spinner: true
  };

  componentWillMount() {
    this.getPosition()
  }

  getPosition() {
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition(
        position => {
          this.getWeatherData(position.coords);
        },
        error => {
          this.setState({
            spinner: false,
            error: error.message
          })
        }, { timeout: 20000, enableHighAccuracy: true }
      );
    } else {
      this.setState({
        spinner: false,
        error: false,
        data: Object.assign(
          {},
          this.state.data,
          { name: 'Geolocation is not supported by this browser.' })
      });
    }
  }

  getWeatherData(location) {
    const { latitude: lat, longitude: lon } = location;
    fetch(`https://fcc-weather-api.glitch.me/api/current?lat=${lat}&lon=${lon}`).then(response =>
      response.json().then(data => {
        this.setState({
          spinner: false,
          data
        });
        return data;
      })
    );
  }

  render() {
    const { spinner, data, error } = this.state;
    return (
      <MuiThemeProvider>
        <React.Fragment>
          {spinner ? (
            <Progress show={spinner} />
          ) : (
            <WeatherContainer weather={data} error={error} />
          )}
        </React.Fragment>
      </MuiThemeProvider>
    );
  }
}
