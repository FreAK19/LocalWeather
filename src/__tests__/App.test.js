import React from 'react';
import fetchMock from 'fetch-mock';
import App from '../App';
import Progress from '../components/Progress/Progress';

describe('App', () => {
  test('should render correctly', () => {
    const component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });

  test('should render Progress component if state.spinner is true', () => {
    const component = shallow(<App />);
    component.setState({ spinner: true });
    expect(component.find(Progress).length).toBe(1);
  });

  test('should not render Progress if state.spinner is false', () => {
    const component = shallow(<App />);
    component.setState({ spinner: false });
    expect(component.find(Progress).length).toBe(0);
  });

  test.skip('getWeatherData method should return json response', () => {
    expect.assertions(1);
    //  const spy = jest.spyOn(App.prototype, 'getWeatherData');
    const component = shallow(<App />);
    fetchMock.mock('https://fake.com', 'GET', {
      any: 'value'
    });
    const location = { latitude: 50, longitude: 30 }; //  only this prop is need
    const response = component.instance().getWeatherData(location);
    const res = response;
    expect(res).toBe(true);
  });
});
