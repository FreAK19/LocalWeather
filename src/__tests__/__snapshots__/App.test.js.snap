// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<MuiThemeProvider>
  <React.Fragment>
    <WeatherContainer
      error={false}
      weather={
        Object {
          "main": Object {
            "humidity": 0,
            "pressure": 0,
            "temp": 0,
            "temp_max": 0,
            "temp_min": 0,
          },
          "name": "Geolocation is not supported by this browser.",
          "weather": Array [
            Object {
              "description": "Weather Status",
              "id": 800,
              "main": "Clear",
            },
          ],
          "wind": Object {
            "speed": 0,
          },
        }
      }
    />
  </React.Fragment>
</MuiThemeProvider>
`;
